ui-admin/src/forms/designer/split/SplitFormDesigner.tsx

```diff
@@ -16,81 +22,78 @@ export const SplitFormDesigner = ({ content, onChange, currentLanguage, supporte
     content: FormContent, onChange: (newContent: FormContent) => void,
     currentLanguage: PortalEnvironmentLanguage, supportedLanguages: PortalEnvironmentLanguage[]
 }) => {
+  const HIDE_TABLE_OF_CONTENTS_KEY = 'formDesigner.hideTableOfContents'
+
   const [currentPageNo, setCurrentPageNo] = useState(0)
-  const [currentPage, setCurrentPage] = useState(content.pages[currentPageNo])
+  const [hideTableOfContents, setHideTableOfContents] = useState(
+    localStorage.getItem(HIDE_TABLE_OF_CONTENTS_KEY) === 'true')
+  const [searchParams, setSearchParams] = useSearchParams()
+  const selectedElementPath = searchParams.get('selectedElementPath') ?? 'pages'
 
-  const handlePageChange = (direction: 'next' | 'previous') => {
-    if (direction === 'next') {
-      setCurrentPageNo(currentPageNo + 1)
-      setCurrentPage(content.pages[currentPageNo + 1])
-    } else {
-      setCurrentPageNo(currentPageNo - 1)
-      setCurrentPage(content.pages[currentPageNo - 1])
-    }
-    handleScrollToTop()
+  const setSelectedElementPath = (path: string) => {
+    searchParams.set('selectedElementPath', path)
+    setSearchParams(searchParams)
+    scrollToElement(path, setCurrentPageNo)
   }
 
-  const handleScrollToTop = () => {
-    window.scrollTo(0, 0)
-  }
+  useEffect(() => {
+    if (selectedElementPath && selectedElementPath !== 'pages') {
+      scrollToElement(selectedElementPath, setCurrentPageNo)
+    }
+  }, [])
 
-  return <div className="mt-3 container w-100">
-    <div className="d-flex">
-      {renderNewElementButton(content, onChange, -1, currentPageNo, 'question')}
-      {renderNewElementButton(content, onChange, -1, currentPageNo, 'panel')}
-    </div>
-    <>
-      {currentPage && currentPage.elements && currentPage.elements.map((element, elementIndex) => (
-        <div key={elementIndex} className="container">
-          <SplitFormElementDesigner currentPageNo={currentPageNo}
-            elementIndex={elementIndex} editedContent={content}
-            element={content.pages[currentPageNo].elements[elementIndex]}
-            currentLanguage={currentLanguage} supportedLanguages={supportedLanguages}
-            onChange={onChange}/>
-          <div className="d-flex">
-            {renderNewElementButton(content, onChange, elementIndex, currentPageNo, 'question')}
-            {renderNewElementButton(content, onChange, elementIndex, currentPageNo, 'panel')}
+  return <div className="container-fluid overflow-scroll">
+    <div className="row w-100 mx-0">
+      <div style={{ overflowY: 'scroll' }}
+        className={classNames('px-0 border-start border-end bg-white', hideTableOfContents ? 'd-none' : 'col-3')}>
+        { !hideTableOfContents &&
+            <FormTableOfContents
+              formContent={content}
+              selectedElementPath={selectedElementPath}
+              onSelectElement={setSelectedElementPath}
+            />
```

Comment:
This might be a controversial opinion, but I find it really hard to map what you're seeing on the right to what's on the left. I think this is an awesome V1 and the functionality is great, but I would like to posit an alternative design idea: a `Go To Question` button.

<img width="1256" alt="image" src="https://github.com/user-attachments/assets/32457542-38c9-4cab-8d0b-259e032879ae">

This go to question button opens a modal with the table of contents (as-is). If you click on a question in the modal, it takes you straight there and scrolls you to it. I think I would personally find that more intuitive and save a ton of screen real estate. I say this because I exclusively use the table of contents for a "bring me straight to this question" purpose, but I could be missing other uses people have.

Reply:
That's an interesting idea- let's keep it in mind as we start to roll this out and gather feedback. I agree that the ToC takes up quite a bit of real estate as-is. Hopefully the ability to hide (with that preference remembered in local storage) can help get it out of the way for when it isn't needed.

We could also roll out both options and let them live side by side for a bit, and then track with mixpanel to see what people are doing.

I think there's also some room to improve the current ToC in the future by adding a "search for stable id", with behavior similar to what you're suggesting.